lisa/target.py

```diff
@@ -350,17 +350,34 @@ def from_one_conf(cls, path):
         return cls.from_conf(conf=conf, plat_info=plat_info)
 
     @classmethod
-    def from_cli(cls, argv=None) -> 'Target':
```

Comment:
What about just having a single `from_cli()` ? It's *slightly* (only slightly, I promise) ugly but if we pass `params=None` then we could just return `target`, otherwise we'd return `args, target`.

Reply:
I'd rather avoid having this kind of ad-hoc polymorphism the wrong way around, it can quickly leads to very strange errors that are very hard to debug. The input data should drive the path taken by the control flow, but not the code structure itself.